packages/x-data-grid/src/components/cell/GridEditSingleSelectCell.tsx

```diff
@@ -1,7 +1,7 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
 import { unstable_useEnhancedEffect as useEnhancedEffect } from '@mui/utils';
-import { SelectProps, SelectChangeEvent } from '@mui/material/Select';
+import { SelectProps, SelectChangeEvent, BaseSelectProps } from '@mui/material/Select';
```

Comment:
How about we do this instead?

```ts
import { SelectProps, SelectChangeEvent } from '@mui/material/Select';

type BaseSelectProps = Omit<SelectProps, 'variant'> & Partial<Pick<SelectProps, 'variant'>>;
```

Would this allow us to unblock the release and avoid bumping the `@mui/material` dependency?

Reply:
Yes, but the core team is close to solve the regression, so I would prefer to have them solving the issue and avoid introducing those typing modification

https://mui-org.slack.com/archives/C0170JAM7ML/p1710415123542469?thread_ts=1709851792.385099&cid=C0170JAM7ML

Reply:
With this solution, we can avoid bumping the version of `@mui/material` if we agree on this – see related discussion in https://github.com/mui/mui-x/pull/12105#discussion_r1524825514